### ISSUE TITLE
Old node modules need cleared out (Cannot read properties of null (reading 'edgesOut')

### ISSUE THREAD

**Issue by katmai** [OpenDevin/OpenDevin#834]
#### Describe the bug
trying to run make build on the latest code and it ends up in this error:

Cannot read properties of null (reading 'edgesOut')

#### Setup and configuration
**Current version**:

```
commit 229fa988c575c291cff6ffc1f9d15814d9d2a884 (HEAD -> main, origin/main, origin/HEAD)
Author: Xingyao Wang <xingyao6@illinois.edu>
Date:   Sun Apr 7 01:04:17 2024 +0800

    remove seed=42 to fix #813 (#830)
```

**My config.toml and environment vars** (be sure to redact API keys):
```
LLM_API_KEY="ollama"
LLM_MODEL="ollama/dolphin-mixtral:latest"
LLM_EMBEDDING_MODEL="local"
LLM_BASE_URL="http://localhost:11434"
WORKSPACE_DIR="./workspace"
```

**My model and agent** (you can see these settings in the UI):
* Model:
* Agent:

**Commands I ran to install and run OpenDevin**:
```
make build
```

**Steps to Reproduce**:
1. pull latest code
2. make build
3.

**Logs, error messages, and screenshots**:
```
142 http fetch GET 200 https://registry.npmjs.org/@swc%2fcore 6ms (cache hit)
143 silly fetch manifest @swc/helpers@^0.5.0
144 http fetch GET 200 https://registry.npmjs.org/@swc%2fhelpers 2ms (cache hit)
145 silly fetch manifest postcss@^8.4.12
146 http fetch GET 200 https://registry.npmjs.org/postcss 6ms (cache hit)
147 silly fetch manifest typescript@>=4.1.0
148 http fetch GET 200 https://registry.npmjs.org/typescript 50ms (cache hit)
149 silly fetch manifest typescript@^4.9.5
150 silly fetch manifest vitest@^0.29.2
151 silly fetch manifest @vitest/browser@*
152 silly fetch manifest vitest@1.4.0
153 silly fetch manifest @types/node@^18.0.0 || >=20.0.0
154 timing idealTree Completed in 4380ms
155 timing command:install Completed in 4385ms
156 verbose stack TypeError: Cannot read properties of null (reading 'edgesOut')
156 verbose stack     at #loadPeerSet (/home/atlas/.nvm/versions/node/v18.20.1/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1313:38)
156 verbose stack     at async #buildDepStep (/home/atlas/.nvm/versions/node/v18.20.1/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:924:11)
156 verbose stack     at async Arborist.buildIdealTree (/home/atlas/.nvm/versions/node/v18.20.1/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:203:7)
156 verbose stack     at async Promise.all (index 1)
156 verbose stack     at async Arborist.reify (/home/atlas/.nvm/versions/node/v18.20.1/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/reify.js:154:5)
156 verbose stack     at async Install.exec (/home/atlas/.nvm/versions/node/v18.20.1/lib/node_modules/npm/lib/commands/install.js:153:5)
156 verbose stack     at async module.exports (/home/atlas/.nvm/versions/node/v18.20.1/lib/node_modules/npm/lib/cli-entry.js:61:5)
157 verbose cwd /home/atlas/OpenDevin/frontend
158 verbose Linux 6.6.4-060604-generic
159 verbose node v18.20.1
160 verbose npm  v10.5.0
161 error Cannot read properties of null (reading 'edgesOut')
162 verbose exit 1
163 timing npm Completed in 4511ms
164 verbose unfinished npm timer reify 1712423688807
165 verbose unfinished npm timer reify:loadTrees 1712423688810
166 verbose unfinished npm timer idealTree:buildDeps 1712423691257
167 verbose unfinished npm timer idealTree:node_modules/.pnpm/@monaco-editor+react@4.6.0_monaco-editor@0.47.0_react-dom@18.2.0_react@18.2.0/node_modules/@monaco-editor/react 1712423692071
168 verbose code 1
169 error A complete log of this run can be found in: /home/atlas/.npm/_logs/2024-04-06T17_14_48_682Z-debug-0.log

```

**Comment by rbren** [comment]
Does running `npm i -D styled-components@latest` fix the issue?

We don't seem to have that as a dep, but the internet says it's the problem...

**Comment by katmai** [comment]
no doesn't look like. i ran it and then "make build" for the whole thing to complete and it's output the same error.

```
atlas@chia01:~/OpenDevin$ npm i -D styled-components@latest

added 22 packages in 5s

4 packages are looking for funding
  run `npm fund` for details

```

**Comment by PierrunoYT** [comment]
Same here

**Comment by JayQuimby** [comment]
I'm having the same issue. Same config and output log as OP.

```
156 verbose cwd /home/quimbo/OpenDevin/frontend
157 verbose Linux 5.15.146.1-microsoft-standard-WSL2
158 verbose node v20.12.1
159 verbose npm  v10.5.0
160 error Cannot read properties of null (reading 'edgesOut')
161 verbose exit 1
```

**Comment by enyst** [comment]
After installing styled-components, I was able to start the backend and frontend separately. Frontend will miss the internationalization, but it seems to work otherwise

**Comment by enyst** [comment]
There is an open node bug: https://github.com/npm/cli/issues/4367

It seems to happen a lot when people switch from pnpm to npm. It looks like a consequence of this https://github.com/OpenDevin/OpenDevin/commit/f40fe6ac2873ebfa33cc8a60b72db98d365f6114

**Comment by zhonggegege** [comment]
> There is an open node bug: [npm/cli#4367](https://github.com/npm/cli/issues/4367)
> 
> It seems to happen a lot when people switch from pnpm to npm. It looks like a consequence of this [f40fe6a](https://github.com/OpenDevin/OpenDevin/commit/f40fe6ac2873ebfa33cc8a60b72db98d365f6114)
> 
> Edit: it's an issue due to modules installed by pnpm previously. What I did: go to frontend folder, clear node_modules folder. Then either go back to run make build from the main folder, or in the frontend folder, do `npm install` `npm run make-i18n`

After clearing the node_modules folder, make build failed. I could only `cd frontend`, then `npm install`, then make build, and finally it worked.

**Comment by enyst** [comment]
Cool, thanks for letting us know!

**Comment by zhonggegege** [comment]
> Oh, it still failed. Try to start separately, make start-backend, then make start-frontend.

The independent startup can be successful, but the error here still occurs in the backend. The configuration of config.toml is invalid. It can still only read the model that has been set by the front end. This will also occur if the configuration file does not match the model name path format set by the front end. "null" case.

logs--------------------------------------------------------------
INFO:     connection open

Provider List: https://docs.litellm.ai/docs/providers

ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/uvicorn/protocols/websockets/websockets_impl.py", line 240, in run_asgi
    result = await self.app(self.scope, self.asgi_receive, self.asgi_send)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 69, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/starlette/middleware/errors.py", line 151, in __call__
    await self.app(scope, receive, send)
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/starlette/middleware/cors.py", line 77, in __call__
    await self.app(scope, receive, send)
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 65, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/starlette/routing.py", line 756, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/starlette/routing.py", line 776, in app
    await route.handle(scope, receive, send)
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/starlette/routing.py", line 373, in handle
    await self.app(scope, receive, send)
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/starlette/routing.py", line 96, in app
    await wrap_app_handling_exceptions(app, session)(scope, receive, send)
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/starlette/routing.py", line 94, in app
    await func(session)
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/fastapi/routing.py", line 348, in app
    await dependant.call(**values)
  File "/home/agent/OpenDevin/opendevin/server/listen.py", line 42, in websocket_endpoint
    await session_manager.loop_recv(sid, agent_manager.dispatch)
  File "/home/agent/OpenDevin/opendevin/server/session/manager.py", line 37, in loop_recv
    await self._sessions[sid].loop_recv(dispatch)
  File "/home/agent/OpenDevin/opendevin/server/session/session.py", line 33, in loop_recv
    await dispatch(action, data)
  File "/home/agent/OpenDevin/opendevin/server/agent/manager.py", line 74, in dispatch
    await self.create_controller(data)
  File "/home/agent/OpenDevin/opendevin/server/agent/manager.py", line 127, in create_controller
    llm = LLM(model=model, api_key=api_key, base_url=api_base)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/agent/OpenDevin/opendevin/llm/llm.py", line 36, in __init__
    self._router = Router(
                   ^^^^^^^
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/litellm/router.py", line 309, in __init__
    self.set_model_list(model_list)
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/litellm/router.py", line 2176, in set_model_list
    deployment = self._add_deployment(deployment=deployment)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/litellm/router.py", line 2212, in _add_deployment
    ) = litellm.get_llm_provider(
        ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/litellm/utils.py", line 5774, in get_llm_provider
    raise e
  File "/home/agent/miniconda3/envs/opendev/lib/python3.11/site-packages/litellm/utils.py", line 5761, in get_llm_provider
    raise litellm.exceptions.BadRequestError(  # type: ignore
litellm.exceptions.BadRequestError: LLM Provider NOT provided. Pass in the LLM provider you are trying to call. You passed model=null
 Pass model as E.g. For 'Huggingface' inference endpoints pass in `completion(model='huggingface/starcoder',..)` Learn more: https://docs.litellm.ai/docs/providers
INFO:     connection closed

**Comment by katmai** [comment]
pulled the newest code and re-ran the build and this seems to have been fixed

```
commit 04066ca42b370fa6d6575c8bcc56dd2ed4210752 (HEAD -> main, origin/main, origin/HEAD)
Author: Engel Nyst <enyst@users.noreply.github.com>
Date:   Sun Apr 7 18:55:52 2024 +0200

    fix default logging (#867)

```

**Comment by rbren** [comment]
For anyone ending up here: try `rm -r frontend/node_modules` before re-installing
